src/btree/col_srch.c

```diff
@@ -90,15 +103,13 @@ restart:	page = current->page;
 		 * page; otherwise return on error, the swap call ensures we're
 		 * holding nothing on failure.
 		 */
-		switch (ret = __wt_page_swap(session, current, descent, 0)) {
-		case 0:
+		if ((ret = __wt_page_swap(session, current, descent, 0)) == 0) {
 			current = descent;
-			break;
-		case WT_RESTART:
-			goto restart;
-		default:
-			return (ret);
+			continue;
 		}
+		if (ret == WT_RESTART)
```

Comment:
We discussed the intricacies of the row search loop on the phone today. I presume the same reasoning applies to the column store case, but it would be good if any comment updates are reflected here as well.

The swap/restart logic is complex enough that it would be nice if column and row store could share it. I know this is a performance path, so I'm not pushing hard for that.